crates/isograph_compiler/src/write_artifacts.rs

```diff
@@ -85,4 +132,10 @@ pub enum GenerateArtifactsError {
         \nReason: {message:?}"
     )]
     UnableToDeleteDirectory { path: PathBuf, message: String },
+
+    #[error(
+        "Unable to write to the persistend documents file. \
+        \nReason: {message:?}"
+    )]
+    UnableToSerializePersistentDocuments { message: String },
```

Comment:
nit: `Persisted`, not `Persistent`, but this error message probably won't exist once the persisted json object is part of the `path_and_contents`